### ISSUE TITLE
False positives when using the typing module and python 3.5.3+

### ISSUE THREAD

**Issue by eddie-dunn** [PyCQA/pylint#1295]
Pylint erroneously reports `invalid-sequence-index` and `not-callable`.

### Steps to reproduce
1. 
```
$ python --version
Python 3.5.3
```
2. Create file `false_positives.py` with the following contents:
```
"""This module should not trigger Pylint errors"""

import typing

# Create named tuple 'Named':
Named = typing.NamedTuple('Named', [('foo', int), ('bar', int)])  # pylint: disable=invalid-name


def foo_the_bar() -> typing.Tuple[str, str]:
    """Function to foo the bar"""
    return ('foo', 'bar')


def create_named_tuple() -> typing.NamedTuple:
    """Create and return a NamedTuple"""
    named = Named(1, 2)
    return named
```
3. 
```
$ pylint -E false_positives.py
No config file found, using default configuration
************* Module false_positives
E:  9, 0: Sequence index is not an int, slice, or instance with __index__ (invalid-sequence-index)
E: 16,12: Named is not callable (not-callable)
```

### Current behavior

`typing.Tuple[type, type]` triggers the `invalid-sequence-index` error.

Instantiating `typing.NamedTuple` objects triggers the `not-callable` error.

### Expected behavior

No such errors should be triggered. The same errors appear with Python 3.6.0. It does not happen when using pylint and Python 3.5.2.

### pylint --version output

```
$ pylint --version
No config file found, using default configuration
pylint 1.6.5, 
astroid 1.4.9
Python 3.5.3 (default, Jan 25 2017, 18:14:52) 
[GCC 4.9.2]

```
